widget/completionentry.go

```diff
@@ -2,21 +2,26 @@ package widget
 
 import (
 	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/driver/desktop"
 	"fyne.io/fyne/v2/theme"
 	"fyne.io/fyne/v2/widget"
 )
 
 // CompletionEntry is an Entry with options displayed in a PopUpMenu.
 type CompletionEntry struct {
 	widget.Entry
-	popupMenu     *widget.PopUp
-	navigableList *navigableList
-	Options       []string
-	pause         bool
-	itemHeight    float32
+
+	Options           []string
+	OnCompleted       func(string) string // Called when you select an item from the list ; return different string to override the completion
+	SubmitOnCompleted bool                // True will call Entry.OnSubmited when you select a completion item
```

Comment:
Are you sure this needs to be optional? The `OnSubmitted` call is already optional, so this is kinda doubled up and could just always on can't it?

Reply:
OnComplete is important to eventually modifiy the string to be inserted in the Entry.

Imagine what is stored in Options is not exactly what should be inserted in Entry...(*)

SubmitOnComplete is for when we want OnSubmitted to be called uppon completion. We could manually call OnSubmitted inside OnComplete but imagine we are in the case above (*), when in OnComplete the Entry content is not yet right.

Reply:
Sorry for letting this drop - I guess I didn't understand the use-case. Both the list and the Entry are user facing, so why should what is input not match what is selected?

p.s. "Completed" is misleading here - It means the action is finished but when you select an item you are not necessarily "done" - Other widgets use different verbs to indicate that a menu item is selected / actioned.